en_US/security/authn/authn.md

```diff
@@ -1,243 +1,812 @@
 # Introduction
 
-Authentication is an important part of most applications. MQTT protocol supports username/password authentication. Enabling authentication can effectively prevent illegal client connections.
+Authentication is an important part of most applications. MQTT protocol supports username/password authentication as
+well as some enhanced methods, like SCRAM authentication. Enabling authentication can effectively prevent illegal client connections.
 
-Authentication in EMQX Broker means that when a client connects to EMQX Broker, the server configuration  is used to control the client's permission to connect to the server.
+Authentication in EMQX Broker means that when a client connects to EMQX Broker, the server configuration is used to control the client's permission to connect to the server.
 
 EMQX Broker's authentication support includes two levels:
 
-- The MQTT protocol specifies the user name and password in the CONNECT packet by itself. EMQX Broker supports multiple forms of authentication based on Username, ClientID, HTTP, JWT, LDAP, and various databases such as MongoDB, MySQL, PostgreSQL, Redis through plugins.
+- The MQTT protocol itself specifies authentication primitives. EMQX Broker supports multiple variants of MQTT-level authentication:
+  * username/password authentication with various backends (MongoDB, MySQL, PostgreSQL, Redis and built-in database);
+  * SCRAM authentication with the built-in database;
+  * JWT authentication;
+  * authentication via custom HTTP API.
 - At the transport layer, TLS guarantees client-to-server authentication using client certificates and ensures that the server verifies the server certificate to the client. PSK-based TLS/DTLS authentication is also supported.
 
-This authentication methods supported by EMQX and the configuration methods of the corresponding plugins are introduced in this article.
-
-## Authentication method
-
-EMQX supports the use of built-in data sources (files, built-in databases), JWT, external mainstream databases, and custom HTTP APIs as authentication data sources.
-
-The data source connection and authentication logic are implemented through plugins. Each plugin corresponds to an authentication method, and the corresponding plugin needs to be enabled before use.
-
-When the client connects, the plugin implements the identity authentication of the client by checking whether its username/clientid and password are consistent with the information of the specified data source.
-
-Authentication methods supported by EMQX:
+In this document, we describe EMQX authentication and its configuration concepts.
+
+## Authentication sources
+
+_Authentication source_ (or simply _authenticator_) is an EMQX module that implements MQTT authentication. The following authenticators are
+available by default:
+
+| mechanism        | backend            | description                                                                   |
```

Comment:
A bit of clarification: _authenticators_ are defined just above, but here we have "mechanisms" and "backends".  Maybe point out how an authenticator maps to these?  For instance, an authenticator maps to a single (mechanism, backend) pair?

Reply:
✅ 

Btw, now it seems for me that http authenticator shouldn't be `password_based` with `http` backend — there is no actual password verification (and http backend even may not check passwords).

Maybe, for http authenticator it would be more reasonable to have mechanism=`http` and do not have backend? 🤔

Reply:
It sounds like HTTP is more akin to `jwt` authn.  But wouldn't `http` be more like the backend rather than the mechanism, since the latter can be something arbitrary for HTTP?  :thinking:

Reply:
M.b. something like `external:http`? With `external` for mechanism and `http` for the backend?

Reply:
Does this imply code/api changes?

Reply:
Anyway, this can be a future improvement. Resolving the comment for now.